File: Assignment_5/Merge.java
class Merge{
    static int[] merge(int[] a, int[] b){
        int[] c = new int[a.length + b.length];
        int i = 0, j = 0, k = 0;
        while(i < a.length && j < b.length){
            if(a[i] < b[j]){
                c[k++] = a[i++];
            }else{
                c[k++] = b[j++];
            }
        }
        while(i < a.length){
            c[k++] = a[i++];
        }
        while(j < b.length){
            c[k++] = b[j++];
        }
        return c;
    }

    public static void main(String[] args){
        int[] a = {1, 3, 5, 7, 9};
        int[] b = {2, 4, 6, 8, 10, 11, 12};
        int[] c = merge(a, b);
        for(int i = 0; i < c.length; i++){
            System.out.print(c[i] + " ");
        }
    }
}

File: Assignment_5/Merge.fs
module Merge

let merge lst1 lst2 = 
    let rec _merge lst1 lst2 acc = 
        match lst1 with
        | [] -> (List.rev acc) @ lst2 
        | x :: xs -> 
            match lst2 with
            | [] -> (List.rev acc) @ lst1
            | y :: ys -> if x <= y then _merge xs lst2 (x :: acc) 
                                   else _merge lst1 ys (y :: acc)
    _merge lst1 lst2 []
